### PR TITLE
fix: run lifecycle hooks outside of sandbox by default

### PR DESCRIPTION
**Pull request by gregmagolan**
At one point while benchmarking I had intended to make this the default. Running outside of the sandbox is a better default for performance since the output TreeArtifact doesn't need to be copied out of the sandbox after the action completes. The symlink node_modules structure already provides strict dependencies as well by virtue of the layout.

This would also prevent issues such as https://github.com/aspect-build/rules_js/issues/386 when some package lc hooks follow symlinks out of the sandbox and fail.

### CODE REVIEW

**Comment by jbedard** [comment]
Is there any way to test this?

File: npm/npm_import.bzl [hunk 1]
```diff
@@ -225,11 +236,15 @@ def npm_import(
             At best omitting this field will make your build non-hermetic.
 
             It is optional to make development easier but should be set before shipping.
+
```
Comment (jbedard) [inline npm/npm_import.bzl:239]: Are these formatting changes correct? The other arguments above the diff don't have the blank line...

File: npm/private/npm_translate_lock.bzl [hunk 2]
```diff
@@ -181,6 +181,17 @@ _ATTRS = {
         }
         """,
     ),
+    "lifecycle_hooks_no_sandbox": attr.bool(
+        doc = """If True, the "no-sandbox" execution requirements is added
+        to all lifecycle hooks.
+
+        Equivalent to adding `"*": ["requires-network"]` to lifecycle_hooks_execution_requirements.
```
Comment (jbedard) [inline npm/private/npm_translate_lock.bzl:188]: Shouldn't this be `"no-sandbox"`?

**Review by jbedard — APPROVED** [review]

File: npm/private/npm_translate_lock.bzl (hunk 2 above)
Comment (gregmagolan) [reply npm/private/npm_translate_lock.bzl:188]: Indeed. Nice catch.

File: npm/npm_import.bzl (hunk 1 above)
Comment (gregmagolan) [reply npm/npm_import.bzl:248]: Whoops. Cleaned up.

**Comment by bhavitsharma** [comment]
Love it! Thanks again for the quick resolution!

**Pull request merged**